### PR TITLE
Introduce Auto-Format to Sprig Editor  #1399

### PR DESCRIPTION
**Pull request by DevIos01**
### **Introduction**
This pull request introduces an Auto-Format feature to the editor, aimed at improving code readability and consistency. The motivation behind this feature is to provide an easy-to-access tool for users, especially beginners and those participating in hackathons, to automatically format their code. This addresses the common issue of inconsistent code styling, such as indentation and spacing, making the codebase cleaner and easier to read.

### **Changes Made**
Added **stripWhitespaceAndUpdate** function to include js-beautify options.
Added a menu item "Prettify code" in the dropdown, linked to the updated function.

### **Before & After**

**Before: Code that's a bit all over the place.**

image

**After: Nicely formatted code that's easy on the eyes.**

image

image

### CODE REVIEW

File: src/components/navbar-editor.tsx [hunk 1]
```diff
@@ -57,6 +58,34 @@ type StuckData = {
...
 
+const stripWhitespaceAndUpdate = () => {
+    if (!codeMirror.value) return;
+
+    const currentText = codeMirror.value.state.doc.toString();
+
+    const beautifyOptions = {
+        indent_size: 2,
+        space_in_empty_paren: true,
+        e4x: true,
```
Comment (JosiasAurel) [inline src/components/navbar-editor.tsx:69]: please add a comment specifying what this option does

File: src/components/navbar-editor.tsx [hunk 2]
```diff
@@ -57,6 +58,34 @@ type StuckData = {
...
+        end_with_newline: true,
+        wrap_line_length: 0,
+        preserve_newlines: false,
+        max_preserve_newlines: 2,
+        jslint_happy: false,
+        brace_style: "end-expand",
```
Comment (JosiasAurel) [inline src/components/navbar-editor.tsx:75]: also this one

File: src/components/navbar-editor.tsx [hunk 3]
```diff
@@ -57,6 +58,34 @@ type StuckData = {
...
+
+    const beautifyOptions = {
+        indent_size: 2,
+        space_in_empty_paren: true,
+        e4x: true,
+        end_with_newline: true,
+        wrap_line_length: 0,
+        preserve_newlines: false,
+        max_preserve_newlines: 2,
+        jslint_happy: false,
```
Comment (JosiasAurel) [inline src/components/navbar-editor.tsx:74]: also this one

**Review by JosiasAurel — CHANGES_REQUESTED** [review]

**Review by JosiasAurel — APPROVED** [review]

**Comment by leomcelroy** [comment]
Look good to me.

**Comment by leomcelroy** [comment]
I reverted this after testing some more. 

Here is the snippet I often use for formatting.

```js
function formatCode(code) {
  try {
    const options = {
      indent_size: 2,
      "brace_style": "collapse,preserve-inline",
    }
    return js_beautify(code, options)
  } catch (error) {
    console.log(error)
    return code // return the original code if there's an error
  }
}
```

**Pull request merged**